js/views/higher-order/withAutoHide.js

```diff
@@ -0,0 +1,194 @@
+const React = require('react');
+const ReactDOM = require('react-dom');
+const CONSTANTS = require('../../constants/constants');
+
+const withAutoHide = function(ComposedComponent) {
+  return class extends React.Component {
+    constructor(props) {
+      super(props);
+
+      this.handleMouseOver = this.handleMouseOver.bind(this);
+      this.hideControlBar = this.hideControlBar.bind(this);
+      this.showControlBar = this.showControlBar.bind(this);
+      this.handleKeyDown = this.handleKeyDown.bind(this);
+      this.startHideControlBarTimer = this.startHideControlBarTimer.bind(this);
+      this.cancelHideControlBarTimer = this.cancelHideControlBarTimer.bind(this);
+      this.handleMouseOut = this.handleMouseOut.bind(this);
+      this.handlePlayerMouseMove = this.handlePlayerMouseMove.bind(this);
+      this.handleTouchEnd = this.handleTouchEnd.bind(this);
+    }
+
+    componentDidMount() {
+      document.addEventListener('mousemove', this.handlePlayerMouseMove, false);
```

Comment:
Is it possible to limit these to the screen's container, or are they needed at the document level?

Reply:
I copied these from its original location in playingScreen.js:

https://github.com/ooyala/html5-skin/blob/master/js/views/playingScreen.js#L47

I'd like to avoid changing the behavior unless necessary.